packages/engine/lib/storage/src/lib.rs

```diff
@@ -1,8 +1,8 @@
-//! Library for low-level to store and load arrow-data.
+//! Defines how data is stored in shared memory and provides an (optional) arrow-interface to it.
```

Comment:
```suggestion
//! Defines how data is stored in shared memory and provides an (optional) Apache Arrow interface to it.
```

I was going to ask if we want to link to the Arrow module but I've noticed we don't have one in the memory crate. Do we think grouping the bits like `ipc` under Arrow isn't helpful?

Reply:
Reading these crate docs make me think we might want an arrow module, as we have memory, IPC, meta, etc. and some of those are arrow specific and some are just raw memory logic.

Reply:
Currently, we have a strict hierarchy of modules in this crate. `memory` has no dependencies, `batch` only depends on `memory`, and `ffi` only depends on the other two. I would keep arrow-related things to this crate for now and moving that into another crate is very easy because of the dependencies we currently have. If we choose to move out arrow, we could use similar schema compilers use (LIR, MIR, HIR) for low-level, mid-level, and high-level things. LIR would be the shared_memory things - an interface to use memory. MIR would be the representation in memory (namely `arrow`), HIR is the actual schemas. "Where to store data" vs "How to store data" vs "What data to store" but I don't think it's required at this state of the separation-task rather than see how it turns out at the end. Currently we are combining the former two questions.

Reply:
Apologies, I'm not suggesting `arrow` should become its own crate, I think that'd be even more confusing as we are literally using an external `arrow` crate. I'm suggesting that it should perhaps be a **module** within this crate so that we better group `ipc` and such under arrow. 

I quite like the idea of breaking things into levels but I agree that perhaps we shouldn't do it right now. I'm just interested in making sure our crate and module structure is intuitive and guides understanding. Right now I don't think it's necessarily clear that some of the modules are directly focused to managing Arrow memory layouts.

Reply:
I'm not proposing naming a crate `arrow` because that would be indeed confusing and annoying. This is mainly why I haven't called the module `arrow`. However, renaming `ipc` to `arrow_ipc` may be suitable. Actually, the main struct provided by `batch` is called `ArrowBatch`, so I don't really have an opinion here 🙃

Reply:
> moving that into another crate is very easy

Sorry I thought that meant creating a crate called `arrow`.

Anyway, I think it's okay to have an `arrow` module. I agree it's a little confusing but I think it's better than not having one.

Reply:
Renamed `batch` to `arrow` in 3d068fe6487f2da7cc2348916abb88f71f1e55ca